### PR TITLE
buildFormTree problem

### PR DESCRIPTION
**Pull request by jglick**
@rsandell found a problem with new HtmlUnit affecting `configRoundTrip` on `CpsFlowDefinition`. @tfennelly said he would take a look.

- [X] reproduce problem
- [x] fix it

@reviewbybees

### CODE REVIEW

**Comment by jglick** [comment]
To see the form content:

```java
diff --git a/src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowDefinition2Test.java b/src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowDefinition2Test.java
index c0dfd89..c720785 100644
--- a/src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowDefinition2Test.java
+++ b/src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowDefinition2Test.java
@@ -60,10 +60,11 @@ public class CpsFlowDefinition2Test extends AbstractCpsFlowTest {
         assertTrue(exec.isComplete());
     }
 
-    @Ignore("TODO fails with new HtmlUnit: buildFormTree produces bogus \"$class\":[\"org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition\",\"org.jenkinsci.plugins.workflow.cps.CpsScmFlowDefinition\"] rather than \"$class\":\"org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition\" as expected")
+    //@Ignore("TODO fails with new HtmlUnit: buildFormTree produces bogus \"$class\":[\"org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition\",\"org.jenkinsci.plugins.workflow.cps.CpsScmFlowDefinition\"] rather than \"$class\":\"org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition\" as expected")
     @Test public void configRoundTrip() throws Exception {
         WorkflowJob job = jenkins.jenkins.createProject(WorkflowJob.class, "p");
         job.setDefinition(new CpsFlowDefinition("echo 'whatever'"));
+        System.out.println(jenkins.createWebClient().getPage(job,"configure").getWebResponse().getContentAsString());
         jenkins.configRoundtrip(job);
     }
 
```

**Comment by jglick** [comment]
Passes if you

```diff
diff --git a/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsScmFlowDefinition.java b/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsScmFlowDefinition.java
index 1e91a41..2e75864 100644
--- a/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsScmFlowDefinition.java
+++ b/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsScmFlowDefinition.java
@@ -131,7 +131,7 @@ public class CpsScmFlowDefinition extends FlowDefinition {
         return System.getProperty(WorkspaceList.class.getName(), "@");
     }
 
-    @Extension public static class DescriptorImpl extends FlowDefinitionDescriptor {
+    /*@Extension*/ public static class DescriptorImpl extends FlowDefinitionDescriptor {
 
         @Inject public Snippetizer snippetizer;
 
```

so somehow related to [this code](https://github.com/jenkinsci/jenkins/blob/67597a7c91ea90dab5b28d17ab2e9a3f9d6439a6/war/src/main/webapp/scripts/hudson-behavior.js#L2473-L2475) running twice when it should only have run once, for the selected entry in [the pulldown](https://github.com/jenkinsci/workflow-job-plugin/blob/8453c79643f252e7b9ceccb9c1ef6f575466ccfe/src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowJob/configure-entries.jelly#L41).

**Comment by jglick** [comment]
I think it is the `<script>` from `Snippetizer` that is to blame.

**Comment by jglick** [comment]
Found the problem; basically the same as in https://github.com/jenkinsci/pipeline-plugin/pull/253.

File: src/main/resources/org/jenkinsci/plugins/workflow/cps/Snippetizer/block.jelly [hunk 1]
```diff
@@ -63,24 +63,24 @@ THE SOFTWARE.
         <f:block>
             <input type="button" value="Generate Groovy" onclick="handlePrototype_${id}(); return false" class="submit-button primary"/>
             <f:textarea id="prototypeText_${id}" readonly="true" style="margin-top: 10px" />
-        </f:block>
-        <script>
-            function handlePrototype_${id}() {
```
Comment (jglick) [inline src/main/resources/org/jenkinsci/plugins/workflow/cps/Snippetizer/block.jelly]: [whitespace-ignoring diff](https://github.com/jenkinsci/workflow-cps-plugin/pull/7/files?w=1)

**Comment by kwhetstone** [comment]
:bee:

**Comment by amuniz** [comment]
:bee:

**Comment by jglick** [comment]
To get this fix in other plugins with a `test`-scoped dependency on `workflow-cps`, use

* `2.2` (of just this plugin: [explanation](https://jenkins.io/blog/2016/04/22/pipeline-2/)), for 1.642.x+ baseline
* `1.14.1-beta-1` (of all Pipeline plugins), for 1.609.x baseline
* `1.4.3-beta-1` (of all Pipeline plugins), for 1.580.x baseline

**Comment by KostyaSha** [comment]
I found this on 12 Jan, and @tfennelly know about this issue ;)

> 1.14.1-beta-1 (of all Pipeline plugins), for 1.609.x baseline

Verified, works on 1.625.3, though blows a lot of jquery js detached 404 errors (hpi version 1.115)

File: src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowDefinition2Test.java [hunk 2]
```diff
@@ -58,4 +59,10 @@
         assertTrue(exec.isComplete());
     }
 
+    @Test public void configRoundTrip() throws Exception {
+        WorkflowJob job = jenkins.jenkins.createProject(WorkflowJob.class, "p");
```
Comment (KostyaSha) [inline src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowDefinition2Test.java]: `jenkins.jenkins` ;(

**Pull request merged**